hal/common/BusInOut.cpp

```diff
@@ -49,32 +52,39 @@ BusInOut::~BusInOut() {
 }
 
 void BusInOut::write(int value) {
+    _mutex.lock();
     for (int i=0; i<16; i++) {
         if (_pin[i] != 0) {
             _pin[i]->write((value >> i) & 1);
         }
     }
+    _mutex.unlock();
 }
 
 int BusInOut::read() {
+    _mutex.lock();
     int v = 0;
     for (int i=0; i<16; i++) {
         if (_pin[i] != 0) {
             v |= _pin[i]->read() << i;
         }
     }
+    _mutex.unlock();
     return v;
 }
 
 void BusInOut::output() {
+    _mutex.lock();
     for (int i=0; i<16; i++) {
         if (_pin[i] != 0) {
             _pin[i]->output();
         }
     }
+    _mutex.unlock();
 }
 
 void BusInOut::input() {
+    _mutex.lock();
```

Comment:
This change will be hard to make for everything at this point.  I'll add a lock guard to mbed rtos and this can be switched over later.  Does that sound alright?